src/opera/pge/rtc_s1/rtc_s1_pge.py

```diff
@@ -695,6 +698,59 @@ def _qa_log_filename(self):
         """
         return self._ancillary_filename() + ".qa.log"
 
+    def augment_measured_parameters(self, measured_parameters):
+        """
+        Override of the augment_measured_parameters() method in Base PGE with an added
+        "preprocessing" step to handle the structure of HDF5 metadata. While GeoTIFF
+        metadata is a flat dictionary, HDF5 metadata is a nested dictionary structure,
+        wherein the variable "keys" can be arbitrarily deep into the structure and
+        the values likewise can be nested dictionaries.
+
+        The preprocessing step in this method selectively flattens the metadata
+        dictionary based on the "paths" provided in the variable keys of the configuration
+        YAML file. The result of this preprocessing is then safely passed to the base
+        method to get the correct structure expected by the Jinja template.
+
+        Parameters
+        ----------
+        measured_parameters : dict
+            The HDF5 metadata from the output product. See get_rtc_s1_product_metadata()
+
+        Returns
+        -------
+        augmented_parameters : dict
+            The metadata fields converted to a list with name, value, types, etc
+        """
+        descriptions_file = self.runconfig.iso_measured_parameter_descriptions
+
+        new_measured_parameters = {}
+
+        if descriptions_file:
+            with open(descriptions_file) as infile:
+                descriptions = yaml.safe_load(infile)
+        else:
+            msg = ('Measured parameters configuration is needed to extract the '
+                   'measured parameters attributes from the RTC-S1 metadata')
+            self.logger.critical(self.name, ErrorCode.ISO_METADATA_DESCRIPTIONS_CONFIG_NOT_FOUND, msg)
+
+        for parameter_var_name in descriptions:
+            key_path = parameter_var_name.split(MEASURED_PARAMETER_PATH_SEPARATOR)
+
+            mp = measured_parameters
+
+            while len(key_path) > 0:
+                try:
+                    mp = mp[key_path.pop(0)]
+                    new_measured_parameters[parameter_var_name] = mp
+                except KeyError as err:
+                    msg = (f'Measured parameters contains no entry for description '
+                           f'key "{err}"')
+                    self.logger.warning(self.name, ErrorCode.ISO_METADATA_NO_ENTRY_FOR_DESCRIPTION, msg)
```

Comment:
Would it make sense to switch this error in the other PGEs that override the augment function to use this error code & log level?

Reply:
It made sense for this PGE since the "standard" measured parameters are a superset of the parameters for "static layer" products, so a single description config could be used for both workflows. But to support that it was important that detection of a missing parameter did not cause a hard crash.

Since this is not necessarily the case for other PGEs like CSLC-S1, I'm not sure its appropriate to use the same logic there.

Reply:
Ah ok that makes sense!